pkg/storage/system_intake_funding_source.go

```diff
@@ -0,0 +1,80 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// UpdateSystemIntakeFundingSources clears and updates the funding sources of a system intake
+func (s *Store) UpdateSystemIntakeFundingSources(ctx context.Context, systemIntakeID uuid.UUID, fundingSources []*models.SystemIntakeFundingSource) ([]*models.SystemIntakeFundingSource, error) {
+	now := s.clock.Now()
+	fmt.Println("Updating system intake funding sources: ", len(fundingSources))
+
+	tx := s.db.MustBegin()
+	defer tx.Rollback()
+	fmt.Println("    Began transaction")
+	deleteFundingSourcesSQL := `
+		DELETE FROM system_intake_funding_sources
+		WHERE system_intake_id = $1;
+	`
+	_, err := tx.Exec(deleteFundingSourcesSQL, systemIntakeID.String())
+
+	fmt.Println("    Delete err: ", err)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, fundingSource := range fundingSources {
+		fmt.Println("        Inserting funding source ", i)
+		if fundingSource != nil {
+			if fundingSource.ID == uuid.Nil {
+				fundingSource.ID = uuid.New()
+			}
+			if fundingSource.CreatedAt == nil {
+				fundingSource.CreatedAt = &now
+			}
+			if fundingSource.UpdatedAt == nil {
+				fundingSource.UpdatedAt = &now
+			}
```

Comment:
Similar question - if we're deleting and re-inserting, it seems the `createdAt` and `updatedAt` would almost always be the same. Perhaps we don't need both?